search/src/TrajectoryUtils.cpp

```diff
@@ -0,0 +1,92 @@
+/*
+ * TrajectoryUtils.cpp
+ *
+ * Created on: Sept 23, 2022
+ *
+ * Helper functions for filtering results.
+ */
+
+#include "TrajectoryUtils.h"
+
+namespace kbmod {
+    
+/* Compute the average distance between two trajectories at the
+   given time steps. Used in duplicate filtering and clustering. */
+double aveTrajectoryDistance(const std::vector<pixelPos>& posA,
+                             const std::vector<pixelPos>& posB)
+{
+    const int num_times = posA.size();
+    assert(posB.size() == num_times);
+
+    double sum = 0.0;
+    for (int i = 0; i < num_times; ++i)
+    {
+        double dx = posA[i].x - posB[i].x;
+        double dy = posA[i].y - posB[i].y;
+        sum += sqrt(dx*dx + dy*dy);
+    }
+    
+    return sum / (double)num_times;
+}
+    
+// Converts a trajRegion result into a trajectory result.
+trajectory convertTrajRegion(const trajRegion& t, float endTime)
+{
+    assert(endTime > 0.0);
+    float scale = 1.0;
+    if (t.depth > 0)
+        scale = std::pow(2.0, static_cast<float>(t.depth));
+
+    trajectory tb;
+    tb.lh = t.likelihood;
+    tb.flux = t.flux;
+    tb.obsCount = t.obs_count;
+    tb.x = t.ix * scale;
+    tb.y = t.iy * scale;
+    tb.xVel = (scale * t.fx - scale * t.ix) / endTime;
+    tb.yVel = (scale * t.fy - scale * t.iy) / endTime;
+    return tb;
+}
+
+std::vector<trajRegion> subdivideTrajRegion(const trajRegion& t)
+{
+    short nDepth = t.depth - 1;
+    std::vector<trajRegion> children(16);
+    float nix = t.ix*2.0;
```

Comment:
I realize this is just copy/paste from the previous implementation, but this is a prime opportunity for using more descriptive variable names.